op-node/rollup/sync/start.go

```diff
@@ -109,6 +109,12 @@ func FindL2Heads(ctx context.Context, cfg *rollup.Config, l1 L1Chain, l2 L2Chain
 		return nil, fmt.Errorf("failed to fetch current L2 forkchoice state: %w", err)
 	}
 
+	// Geth on bad restarts can rewind the head back to last valid state, but fails to rewind the finalization or safe marker.
```

Comment:
I think we may still need to do the walk back from this point.

Reply:
good point, the if the rewind is short, but still longer than the distance to finalized head, then it'll hit. A regular reorg should (tm) have proper finalized label change though. I'll change this PR to draft and investigate a geth fix.